### PR TITLE
[aptos-cli] Support vector<vector<u8>> for run function and script

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
A widely requested feature, supporting `vector<vector<u8>>`

### Test Plan

I modified hello blockchain so that it would take a vec vec u8

You can see the resource here:
https://explorer.aptoslabs.com/account/0xfdf0c99777dd2821101c0b1c26a140c8fcfd63f99a371c4103397ed27791e65d?network=Devnet
```
$ aptos -- move run --function-id default::message::set_message --args hex_array:AB,BC
Do you want to submit a transaction for a range of [24800 - 37200] Octas at a gas unit price of 100 Octas? [yes/no] >
yes
{
  "Result": {
    "transaction_hash": "0xc3c3844c923fa9a019dd6e8622883239365440cb71b9377498e483b4e2f379b7",
    "gas_used": 257,
    "gas_unit_price": 100,
    "sender": "fdf0c99777dd2821101c0b1c26a140c8fcfd63f99a371c4103397ed27791e65d",
    "sequence_number": 1,
    "success": true,
    "timestamp_us": 1664214700463742,
    "version": 17886781,
    "vm_status": "Executed successfully"
  }
}

```

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4542)

### CODE REVIEW

**Review by jjleng — APPROVED** [review]
It is better if we can generalize the array type to support other types. e.g. an array of u8s.

**Comment by gregnazario** [comment]
> It is better if we can generalize the array type to support other types. e.g. an array of u8s.

That's a little harder.  But, right now this is the most common use case missing with our current framework contracts

File: crates/aptos/src/move_tool/mod.rs [hunk 1]
```diff
@@ -851,6 +852,18 @@ impl FunctionArgType {
             FunctionArgType::Hex => bcs::to_bytes(
                 &hex::decode(arg).map_err(|err| CliError::UnableToParse("hex", err.to_string()))?,
             ),
+            FunctionArgType::HexArray => {
+                let mut encoded = vec![];
+                for sub_arg in arg.split(',') {
+                    encoded.push(hex::decode(sub_arg).map_err(|err| {
+                        CliError::UnableToParse(
+                            "hex_array",
+                            format!("Failed to parse hex array: {:?}", err.to_string()),
```
Comment (banool) [inline crates/aptos/src/move_tool/mod.rs:861]: Can you explain more here, like "expected args separated by commas"

**Review by banool — APPROVED** [review]

File: crates/aptos/src/move_tool/mod.rs (hunk 1 above)
Comment (gregnazario) [reply crates/aptos/src/move_tool/mod.rs:861]: Yes, I want to get this out, and then we need better documentation here in general around all the input types

**Pull request merged**